pkgs/development/python-modules/jaxlib/default.nix

```diff
@@ -268,6 +268,7 @@ buildPythonPackage {
 
   nativeBuildInputs = lib.optional cudaSupport addOpenGLRunpath;
 
+  # Note that cudatoolkit is necessary since jaxlib looks for "ptxas" in $PATH.
```

Comment:
This should really be patched in, `propagatedBuildInputs` won't propagate cudatoolkit PATH because of `strictDeps = true;`

Reply:
I'm not familiar with `strictDeps`. What does it do exactly? I looked through [the manual](https://nixos.org/manual/nixpkgs/stable/) but it doesn't have a clear definition anywhere.

Reply:
I don't think this would be an easy patch to say the least. The problem code is part of tensorflow and there's lots of places where ptxas is referenced: https://cs.github.com/?q=repo%3Atensorflow%2Ftensorflow+ptxas&p=1&pt=9f8f0af67fa730b51fa54e5d174510575e3126bad9e80d749afa326127e0e59f64151e1aa0e68172d5fa8c55c5c439ccd4bcf6456e957391b0d801618432f20a234e2642a839e083eac00d697a2cc15d198841aa3969a1f10e5d7db932ad76e9ea0d17f78b9738f1&scope=&scopeName=All+repos

Reply:
> This should really be patched in, propagatedBuildInputs won't propagate cudatoolkit PATH because of strictDeps = true;

I'm a strictDeps noob, but I can confirm that something does end up propagating since my test script passes with this PR applied.

Reply:
Oh, we are using a .whl for jaxlib. Look slike the reference to `ptxas` is in the binary... :/

Reply:
Correct, it's a little messy due to the way jaxlib's upstream build system is set up.

Reply:
@jonringer Where do you stand on this atm? Is there a better way to do this or should I proceed with the current approach?

Reply:
This will work in a) the building of dependents and b) when using `nix-shell -p pkgs-that-depends-on-jaxlib` however it will fail with `python3.withPackages`. It works in a) and b) because you have a build environment where `$PATH` is set up. It will fail with `python3.withPackages` because the constructed environment contains only Python packages, which `cudatoolkit` is not.

This is therefore not the right solution. Preferred solution is to patch in the references. If that is not possible, create a symlink in `$out` of `jaxlib` to the executable of interest.